### PR TITLE
CrChron Epispde 2

### PR DESCRIPTION
**Pull request by petele**
What's changed, or what was fixed?
- Adds Chromium Chronicle Episode 2

**Target Live Date:** 2019-05-21

- [ ] This has been reviewed and approved by (NAME)
- [X] I have run `npm test` locally and all tests pass.
- [X] I have added the appropriate `type-something` label.
- [X] I've staged the site and manually verified that my content displays correctly.

**CC:** @petele, @enegiar

Note: this is @enegiar's work, I just created the PR for him.

### CODE REVIEW

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  }
+</style>
+
+# The Chromium Chronicle: Fighting Test Flakiness {: .page-title }
+
+**Episode 2:** May 2019
+
+*by Vasilii in Munich*
+
+[Flaky Tests][flaky-tests-context]are a common problem in Chrome. They
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Misssing space between 'Flaky Tests' and 'are'

Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Flaky Tests, tests should be lower case.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+impact the productivity of other developers, and get disabled over time. 
+Disabled tests mean diminishing test coverage. 
+
+**Triaging Stage**
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: This should probably be an H2:

```
## Triaging stage
```

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+**The OWNERS of directories are responsible for fixing their flaky tests.**
+If you received a bug about a flaky test, spend a few minutes and
+**comment what went wrong** on the bug. If you have an old flaky test and
+it's unclear what went wrong, **try to simply re-enable** the test.
+**Reassign the bug ASAP** if it's clearly a problem in another component.
+The owners of that component should have better judgement about the failure, 
+
+**Debugging Stage:**
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: This should probably be an H2:
```
## Debugging stage
```

Note, it's sentence case and doesn't include the colon at the end.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+A number of [command-line flags][useful-command-lines] are useful for
+fixing flaky tests. For example, **--enable-pixel-output-in-tests**
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Code, command line flags, and other "technical stuff" should be wrapped in backticks (`), so it's shown in the code font. Example:
```
For example, `--enable-pixel-output-in-tests`
```

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+will render the actual browser UI. 
+
+**Have fallback tools** if the debugger makes flakiness disappear. It's
+possible that, under debugger, the test is never flaky. In that case, log
+statements or base::debug::StackTrace can be handy.
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: base::debug::StackTrace should be wrapped in backticks.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+Keep in mind common reasons for EXPECT__* failures besides bugs in production
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: EXPECT__* - should that be formatted in some way? Looks odd as it is...

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+code:
+
+* **Incorrect expectations** (e.g. secure page means HTTPS; it can be a localhost instead).
+* Race conditions due to tests not **waiting for the proper event.**
+
+[Don't test the implementation][not-implementation] but the behavior.
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: This is a negative pattern right?It should probably use the `{ .compare-worse }` tag.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+```
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Nit/Best Practice: include the language you're using here...

Example:  ` ```cpp`

A list of supported languages is at https://github.com/google/code-prettify#how-do-i-specify-the-language-of-my-code

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+// It takes 2 round trips between the UI and the background thread to complete.
+SyncWithTheStore();
+SyncWithTheStore();
+CheckTheStore();
+``` 
+The two round trips may change into three in the future, making the test flaky.
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Best Practice: Leave an extra line between code fences and the next paragraph

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+The following is a correct fix:
+{: .compare-better }
+
+<pre class="prettyprint cc-good lang-cpp">
+SubmitTestPasswordForm();
+WaitUntilCredentialPromptVisible(); 
+</pre>
+
+The fix above is correct under the assumption that
+WaitUntilCredentialPromptVisible() doesn’t actually check the UI.
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: WaitUntilCredentialPromptVisible() should be wrapped in backticks

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+**The browser tests should not depend on external UI events** like "focus lost"
+or “window became foreground”. Imagine an implementation where the prompt
+appears only when the browser window is active. Such an implementation
+would be correct; however, checking for the actual window makes the test flaky.
+
+**Post-fix Stage**
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Should be an H2

File: src/content/en/updates/2019/05/chromium-chronicle-2.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+  }
+</style>
+
+# The Chromium Chronicle: Fighting Test Flakiness {: .page-title }
+
+**Episode 2:** May 2019
+
+*by Vasilii in Munich*
+
+[Flaky Tests][flaky-tests-context] are a common problem in Chrome. They
```
Comment (petele) [inline src/content/en/updates/2019/05/chromium-chronicle-2.md]: Flaky Tests, "tests" should be lower case.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md (hunk 6 above)
Comment (enegiar) [reply src/content/en/updates/2019/05/chromium-chronicle-2.md]: Doesn't look like it, going by the CotT article? https://docs.google.com/document/d/1r4kSxX1Pah0EK5-352zoYw4d4RpJXeIsnDu-rWJHbdc/edit?ts=5cc352d3#

We can confirm with vasilii@ though.

File: src/content/en/updates/2019/05/chromium-chronicle-2.md (hunk 9 above)
Comment (enegiar) [reply src/content/en/updates/2019/05/chromium-chronicle-2.md]: Done!

**Pull request merged**